junit5/src/main/java/sunstone/core/AzureArmTemplateCloudDeploymentManager.java

```diff
@@ -60,17 +45,26 @@ Set<String> getUsedRG() {
     /**
      * Returns resource group name as resources are supposed to share the lifecycle
      */
-    String deploy(String template, Map<String, String> parameters) throws IOException {
-        String deploymentName = "SunstoneDeployment-" + UUID.randomUUID().toString().substring(0,5);
+    void deploy(String template, Map<String, String> parameters, String group, String regionStr) throws IOException {
+        String deploymentName = "SunstoneDeployment-" + UUID.randomUUID().toString().substring(0, 5);
+        Region region = Region.fromName(regionStr);
+        if (region == null) {
+            throw new IllegalArgumentException("Unknown region " + regionStr);
+        }
+
+        if (!armManager.resourceGroups().contain(group)) {
+            armManager.resourceGroups().define(group)
+                    .withRegion(region)
+                    .create();
+        }
```

Comment:
In case that group exists it might make sense to log that this group will be re-used (at least on debug level)

Reply:
Good point. I would even warn the user - the group is deleted in `AfterAllCallback`.

Reply:
Fixed.